projects/transformers/experiments/bertitos.py

```diff
@@ -47,14 +47,23 @@
 # |--------------------------------------------------------------------------|
 # | model           | num params | train loss  | eval loss      | perplexity |
 # |-----------------|:----------:|:-----------:|:--------------:|:----------:|
-# | small_bert_100k | 27,523,072 | 2.951       | 3.861          | 14.532     |
-# | small_bert_50k  | 27,523,072 | 3.373       | 4.417          | 21.386     |
-# | mini_bert_100k  | 10,615,808 | 3.669       | 4.773          | 27.342     |
-# | mini_bert_50k   | 10,615,808 | 4.464       | 5.815          | 56.309     |
-# | tiny_bert_100k  |  4,124,928 | 4.566       | 5.802          | 55.79276   |
-# | tiny_bert_50k   |  4,124,928 | 5.990       | 8.367          | 330.234    |
+# | small_bert_100k | 27,523,072 | 2.951       | 2.676          | 14.532     |
+# | small_bert_50k  | 27,523,072 | 3.373       | 3.063          | 21.386     |
+# | mini_bert_100k  | 10,615,808 | 3.669       | 3.308          | 27.342     |
+# | mini_bert_50k   | 10,615,808 | 4.464       | 4.031          | 56.309     |
+# | tiny_bert_100k  |  4,124,928 | 4.566       | 4.021          | 55.793     |
+# | tiny_bert_50k   |  4,124,928 | 5.990       | 5.800          | 330.234    |
 # |--------------------------------------------------------------------------|
 #
+# Mixed precision training:
+#
+# |------------------------------------------------------------------|
+# | model                | train loss  | eval loss      | perplexity |
+# |----------------------|:-----------:|:--------------:|:----------:|
+# | small_bert_100k_fp16 | 2.964       | 3.891          | 14.844     |
+# | small_bert_50k_fp16  | 3.286       | 4.298          | 19.671     |
```

Comment:
Small impact with fp16, that is good. And training time is 40% lower. 

Eval loss seems to be off, it seems you updated the ones above from base 2 to base e, but forgot to update the fp16 runs.

Reply:
Good eye.